lib/spack/spack/cmd/__init__.py

```diff
@@ -143,15 +150,34 @@ def parse_specs(args, **kwargs):
     except spack.spec.SpecParseError as e:
         msg = e.message + "\n" + str(e.string) + "\n"
         msg += (e.pos + 2) * " " + "^"
-        raise SpackError(msg)
+        err_msg = msg
 
     except spack.spec.SpecError as e:
-
         msg = e.message
         if e.long_message:
             msg += e.long_message
-
-        raise SpackError(msg)
+        err_msg = msg
+
+    if err_msg:
```

Comment:
Needs a comment that says this `if` is about reading files when spec parsing fails